src/Modules/CalcSetup.lua

```diff
@@ -973,21 +1008,16 @@ function calcs.initEnv(build, mode, override, specEnv)
 								gemInstance.supportEffect = supportEffect
 							end
 							if gemInstance.gemData then
-								for _, value in ipairs(propertyModList) do
-									local match = true
-									if value.keywordList then
-										for _, keyword in ipairs(value.keywordList) do
-											if not calcLib.gemIsType(supportEffect.gemData, keyword) then
-												match = false
-												break
-											end
-										end
-									elseif not calcLib.gemIsType(supportEffect.gemData, value.keyword) then
-										match = false
-									end
-									if match then
-										supportEffect[value.key] = (supportEffect[value.key] or 0) + value.value
-									end
+								local playerItems = env.player.itemList
+								local socketedIn = playerItems[groupCfg.slotName] and playerItems[groupCfg.slotName].sockets and playerItems[groupCfg.slotName].sockets[gemIndex]
+								applyGemMods(supportEffect, socketedIn and getGemModList(env, groupCfg, socketedIn.color, gemIndex) or propertyModList)
+								if not processedSockets[gemInstance] then
+									processedSockets[gemInstance] = true
+									applySocketMods(env, gemInstance.gemData, groupCfg, gemIndex, playerItems[groupCfg.slotName] and playerItems[groupCfg.slotName].name)
+									-- Keep track of the count of gems of each color socketed in this group
+									groupCfg["intelligenceGems"] = (groupCfg["intelligenceGems"] or 0) + (gemInstance.gemData.tags["intelligence"] and 1 or 0)
+									groupCfg["dexterityGems"] = (groupCfg["dexterityGems"] or 0) + (gemInstance.gemData.tags["dexterity"] and 1 or 0)
+									groupCfg["strengthGems"] = (groupCfg["strengthGems"] or 0) + (gemInstance.gemData.tags["strength"] and 1 or 0)
```

Comment:
```suggestion
									groupCfg.intelligenceGems = (groupCfg.intelligenceGems or 0) + (gemInstance.gemData.tags.intelligence and 1 or 0)
									groupCfg.dexterityGems = (groupCfg.dexterityGems or 0) + (gemInstance.gemData.tags.dexterity and 1 or 0)
									groupCfg.strengthGems = (groupCfg.strengthGems or 0) + (gemInstance.gemData.tags.strength and 1 or 0)
```
This doesn't need to be a key it can just be groupCfg.intelligenceGems, etc
This is duplicated elsewhere